### PR TITLE
Module2 tasks

### PR DESCRIPTION
**Pull request by preeti-bharti**
Please review

### CODE REVIEW

File: src/app/features/courses/course-list/course-list.component.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+         <div class="col-sm-6">
+         <p><b>{{course.title}}</b></p>
+         <p>{{course.description}}</p>
+     </div>
+     <div class="col-sm-6">
+        <p><b>Authors:</b>&nbsp;{{course.authores}}</p>
+        <p><b>Duration:</b>&nbsp;{{transformDurationToHour(course.duration)}}</p>
+        <p><b>Created:</b>&nbsp;{{course.creationDate | date:'dd.MM.YYY'}}</p>
+        <br>
+        <!-- <ng-content select="[editButton]"></ng-content> -->
```
Comment (pramodbavchikar) [inline src/app/features/courses/course-list/course-list.component.html]: no need of keeping commented code, Please remove

File: src/app/features/courses/course-list/course-list.component.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    <div class="row">
+         <div class="col-sm-6">
+         <p><b>{{course.title}}</b></p>
+         <p>{{course.description}}</p>
+     </div>
+     <div class="col-sm-6">
+        <p><b>Authors:</b>&nbsp;{{course.authores}}</p>
+        <p><b>Duration:</b>&nbsp;{{transformDurationToHour(course.duration)}}</p>
+        <p><b>Created:</b>&nbsp;{{course.creationDate | date:'dd.MM.YYY'}}</p>
+        <br>
```
Comment (pramodbavchikar) [inline src/app/features/courses/course-list/course-list.component.html]: never use br, remove and try to update your code and css

File: src/app/features/courses/courses.component.html [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+<app-header>
+        <app-login></app-login>
+</app-header>
+<div *ngIf="IsCoursesExists(courses)">
+<app-course-list [courseList]="courses">
+    <p id="p_list" list style="font-size:28px">Show list</p>
+    <p  editButton><app-button [showCourse]="ShowCourse"></app-button></p>
+</app-course-list>
+</div>
+<div  id="emptyList"*ngIf="false">
```
Comment (pramodbavchikar) [inline src/app/features/courses/courses.component.html]: no indentation

File: src/app/features/login/login.component.css [hunk 4]
```diff
@@ -0,0 +1,28 @@
+.btnLogout{
+    display: inline-flexbox;
+    background-color:chocolate;
+    border: none;
+    color: white;
+    padding: 5px 30px;
+    text-align: right;
+    font-size: 16px;
+    float: right;
+    margin-top: 10px;
```
Comment (pramodbavchikar) [inline src/app/features/login/login.component.css]: these 4 lines can be reduced to single line with margin:10px 10px 15px 8px;

File: src/app/features/login/login.component.css [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+}
+.UserName{
+    display: inline-flexbox;
+    border: none;
+    color: rgb(10, 6, 6);
+    padding: 5px 20px;
+    text-align: right;
+    font-size: 16px;
+    float: right;
+    margin-top: 10px;
```
Comment (pramodbavchikar) [inline src/app/features/login/login.component.css]: same code can be reduced with creation of another class and use it here

File: src/app/features/login/login.component.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.css']
+})
+export class LoginComponent implements OnInit {
+  User:string="Dave";
```
Comment (pramodbavchikar) [inline src/app/features/login/login.component.ts:12]: What is Dave?

File: src/app/features/courses/course-list/course-list.component.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
+import { Component, OnInit,Input } from '@angular/core';
```
Comment (pramodbavchikar) [inline src/app/features/courses/course-list/course-list.component.ts:1]: no spec file for this?

File: src/app/features/courses/courses.component.ts [hunk 8]
```diff
@@ -0,0 +1,53 @@
+import { Component, OnInit } from '@angular/core';
```
Comment (pramodbavchikar) [inline src/app/features/courses/courses.component.ts:1]: add spec file

**Review by pramodbavchikar — COMMENTED** [review]
Submitted all the comments

File: src/app/features/login/login.component.ts (hunk 6 above)
Comment (preeti-bharti) [reply src/app/features/login/login.component.ts:12]: Dave is logged in user. For module 2 we just added the name to show on UI.

**Pull request merged**